apple/internal/rule_factory.bzl

```diff
@@ -525,6 +525,13 @@ use only extension-safe APIs.
 A list of `.h` files that will be publicly exposed by this framework. These headers should have
 framework-relative imports, and if non-empty, an umbrella header named `%{bundle_name}.h` will also
 be generated that imports all of the headers listed here.
+""",
+            ),
+            "custom_umbrella_header": attr.label(
+                allow_single_file = [".h"],
+                doc = """
+A single `.h` file to use in place of generating an umbrella header for the framework. If this is not
+provided a default header will be used instead.
```

Comment:
"A single .h file to use as the umbrella header for this framework. Usually, this header will have the same name as this target, so that clients can load the header using the `#import <MyFramework/MyFramework.h>` format. If this attribute is not specified, an umbrella header will be generated under the same name as this target."